New serverless pattern - adobe-eventbridge-lambda (#2325)
aws-samples/serverless-patterns: adobe-eventbridge-lambda/adobe-eventbridge-lambda/README.md

> # Adobe Amazon EventBridge Partner Integration to AWS Lambda

Reviewer: ```suggestion
# Send Adobe I/O Events to AWS Lambda through integration with Amazon EventBridge
```

> This pattern demonstrates how to use the Adobe I/O Events Amazon EventBridge Integration and AWS Lambda to process/route events from Adobe. This pattern is leveraging the Adobe I/O Evenets Amazon EventBridge integration to send order events from the customer's Adobe account to their AWS account, via an Amazon EventBridge Partner event bus. Once the Adobe I/O events are in the customer's account, an Amazon EventBridge rule routes updated product events to a downstream Lambda function. In production cases, the Lambda function could transform the event, send it to a downstream application, archive it in a warehouse service, or send a notification email to a customer using SES. Amazon CloudWatch Log Groups are provisioned for debugging and auditing. This pattern deploys two EventBridge rules, one Lambda function, and two CloudWatch Log Groups.

Reviewer: ```suggestion
This pattern demonstrates how to use the Adobe I/O Events integration with Amazon EventBridge to process/route events from Adobe with AWS Lambda. This pattern is leveraging the Adobe I/O Events integration with Amazon EventBridge to send order events from the customer's Adobe account to their AWS account, via an Amazon EventBridge Partner event bus. Once the Adobe I/O events are in the customer's account, an Amazon EventBridge rule routes updated product events to a downstream Lambda function. In a real-world use case, the Lambda function could transform the event, send it to a downstream application, archive it in a warehouse service, or send a notification email to a customer using SES. Amazon CloudWatch log groups are provisioned for debugging and auditing. This pattern deploys two EventBridge rules, one Lambda function, and two CloudWatch log groups.
```

    
Learn more about this pattern at Serverless Land Patterns: https://serverlessland.com/patterns/adobe-eventbridge-lambda

Important: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.

## Requirements

* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
* [AWS CDK CLI](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html) (AWS CDK) installed

> * [Create an Adobe Developer Account](https://www.adobe.com/sign/developer-form.html) if you do not already have one and log in. 

Reviewer: Please briefly mention if it is free and if so, under which conditions

    ```
    cdk deploy --parameters adobeEventBusName=SAAS_EVENT_BUS_NAME_HERE
    ```

4. Note the outputs from the CDK deployment process. These contain the resource names and/or ARNs which are used for testing. This stack will output the name of the Lambda function deployed for testing to the CLI. See the example below. 

```
Outputs:
AdobeIntegrationStack.AdobeProcessSpecificEventsLambdaOutput = AdobeIntegrationStack-AdobeProcessOrderEventsLambd....
```

## How it works

This service interaction uses an existing Adobe SaaS integration in the customer's AWS account. If you do not have the Adobe SaaS integration set up in your AWS account, please set it up before deploying this pattern. View the [integration on Adobe's developer site](https://developer.adobe.com/events/docs/guides/amazon_eventbridge/).

This pattern demonstrates how to:
1. Write EventBridge rules that match Adobe's event pattern
2. Send events from Adobe's EventBridge integration to Amazon CloudWatch for logging and debugging

> 3. Transform Adobe I/O events using AWS Lambda and allows you to connect to other services going forward

Reviewer: ```suggestion
3. Transform Adobe I/O events using AWS Lambda and how to connect to other services 
```
